File: LeetCode/LeetCode Practice Problems/1309. Decrypt String from Alphabet to Integer Mapping.cpp
class Solution {
public:
    string freqAlphabets(string s) {
        string ans = "";
        int n = s.length();
        for(int i=0;i<n;)
        {
            if(i+2<n && s[i+2] == '#')
            {
                int v = (s[i] - '0')*10 + (s[i+1] - '0') - 1;
                char ch = 'a' + v;
                ans+=ch;
                i+=3;
            }
            else
            {
                int v = s[i] - '0' - 1;
                char ch = 'a' + v;
                ans+=ch;
                i++;
            }
        }
        return ans;
    }
};

File: LeetCode/LeetCode Practice Problems/1309-Decrypt String from Alphabet to Integer Mapping.java
class Solution {
    public String freqAlphabets(String s) {
        int n = s.length();
        char ch[] = s.toCharArray();
        String ans = "";
        
        for(int i=0;i<n;) {
            if(i+2 < n && ch[i+2] == '#') {
                int val = (ch[i] - '0')*10 + (ch[i+1] - '0') - 1;
                ans = ans + (char)(val + 'a');
                i += 3;
            } else {
                int val = (ch[i] - '0') - 1;
                ans = ans + (char)(val + 'a');
                i++;
            }
        }
        
        return ans;
    }
}
